pkg/apis/provisioning/v0alpha1/types.go

```diff
@@ -219,14 +219,20 @@ type ResourceWrapper struct {
 	// Path to the remote file
 	Path string `json:"path,omitempty"`
 
-	// The commit hash (if exists)
+	// The request ref (or branch if exists)
 	Ref string `json:"ref,omitempty"`
 
 	// The repo hash value
 	Hash string `json:"hash,omitempty"`
 
+	// Basic repository info
+	Repository ResourceRepositoryInfo `json:"repository"`
+
+	// Typed links for this file (only supported by external systems, github etc)
+	URLs *ResourceURLs `json:"urls,omitempty"`
+
 	// The modified time in the remote file system
-	Timestamp *metav1.Time `json:"timestamp,omitempty"`
+	TimestampMillis int64 `json:"timestampMillis,omitempty"`
```

Comment:
I think this change it's not worth the time in this PR. I believe our APIs should start returning string formatted timestamps. 

Your change would also be incomplete as you didn't update other timestamp spec fields with this convention.